x-pack/plugins/security_solution/public/resolver/store/data/reducer.test.ts

```diff
@@ -48,5 +90,127 @@ describe('Resolver Data Middleware', () => {
+      }
+    });
+  });
+
+  describe('when data was received and stats show more related than the API can provide', () => {
+    let firstChildNodeInTree: TreeNode;
+    let statsForFirstChild: Record<string, number>;
+    let categoryToOverCount: string;
+    beforeEach(() => {
+      const generator = new EndpointDocGenerator('seed');
+      const baseTree = generator.generateTree({
+        ancestors: 1,
+        generations: 2,
+        children: 3,
+        percentWithRelated: 100,
+        alwaysGenMaxChildrenPerNode: true,
+      });
+      const { children } = baseTree;
+      firstChildNodeInTree = [...children.values()][0];
+      statsForFirstChild = firstChildNodeInTree.relatedEvents.reduce(
+        (counts: Record<string, number>, relatedEvent) => {
+          for (const category of [relatedEvent.event.category].flat()) {
+            if (!categoryToOverCount) {
+              categoryToOverCount = category;
+            }
+            counts[category] = counts[category] ? counts[category] + 1 : 1;
+          }
+          return counts;
+        },
+        {}
+      );
+      statsForFirstChild[categoryToOverCount] = statsForFirstChild[categoryToOverCount] + 1;
+
+      const tree = mockResolverTree({
+        events: baseTree.allEvents,
+        cursors: {
+          childrenNextChild: 'aValidChildCursor',
+          ancestryNextAncestor: 'aValidAncestorCursor',
+        },
+        children: [...baseTree.children.values()].map((node) => {
+          const childNode: Partial<ResolverChildNode> = node;
+          childNode.entityID = node.id;
+          if (node.id === firstChildNodeInTree.id) {
+            // attach stats
+            childNode.stats = {
+              events: { total: 1, byCategory: statsForFirstChild },
+              totalAlerts: 0,
+            };
+          }
+          return childNode;
+        }) as ResolverChildNode[],
+      });
+      if (tree) {
+        const action: DataAction = {
+          type: 'serverReturnedResolverData',
+          payload: {
+            result: tree,
+            databaseDocumentID: '',
+          },
+        };
+        store.dispatch(action);
+        const relatedAction: DataAction = {
+          type: 'serverReturnedRelatedEventData',
+          payload: {
+            entityID: firstChildNodeInTree.id,
+            events: firstChildNodeInTree.relatedEvents,
+            nextEvent: 'aValidNextEventCursor',
+          },
+        };
+        store.dispatch(relatedAction);
+      }
+    });
+    it('should have the correct related events', () => {
+      const selectedEventsByEntityId = selectors.relatedEventsByEntityId(store.getState());
+      const selectedEventsForFirstChildNode = selectedEventsByEntityId.get(firstChildNodeInTree.id)!
+        .events;
+
+      expect(selectedEventsForFirstChildNode).toBe(firstChildNodeInTree.relatedEvents);
+    });
+    it('should indicate the limit has been exceeded when the actual count < stats count', () => {
```

Comment:
something about this logic seems backwards to me.  `should indicate the limit has been exceeded when the actual count < stats count'`

Reply:
https://github.com/elastic/kibana/pull/70097/commits/fa4543bb574b8ff253baf21b5da7744541b5a95b updated the wording for clarity